### PR TITLE
fix: resolve TypeScript ESM .js extension imports to .ts source files

### LINKED ISSUE #1503: TypeScript  extension imports fail to resolve call edges
GitNexus fails to track caller/callee relationships when TypeScript files use `.js` extensions in ESM imports.

## Repro Steps

1. Clone a TypeScript ESM repo that uses `.js` extensions in imports, e.g.:
   ```bash
   git clone https://github.com/anthropics/lossless-claw-enhanced.git
   cd lossless-claw-enhanced
   ```
2. Index it:
   ```bash
   gitnexus analyze
   ```
3. Query impact for a function imported with `.js` extension:
   ```bash
   gitnexus impact --repo lossless-claw-enhanced estimateTokens
   ```
4. Observe 0 callers returned.
5. Verify with grep that callers exist:
   ```bash
   grep -r 'estimateTokens' src/ --include='*.ts'
   ```
   Output shows 6+ files calling the function.

## Expected
`impact` / `context` shows callers from `assembler.ts`, `compaction.ts`, `engine.ts`, `db/migration.ts`, etc.

## Actual
`impact` returns 0 callers. `context` shows the symbol exists but no `CALLS` edges.

## Environment
- gitnexus@1.6.4
- Windows 11 / Node 24.15.0
- Repository: `lossless-claw-enhanced` (TypeScript, ESM)

## Root cause hypothesis
The import resolver does not map `.js` extension back to `.ts` source file, so the call edge is dropped during ingestion.

### PR DESCRIPTION
**Pull request by chouzz**
## Summary

Fixes #1503 — TypeScript ESM imports with `.js` extensions now correctly resolve to `.ts` source files.

## Problem

TypeScript ESM (`moduleResolution: node16/bundler`) requires imports to use `.js` extensions:
```typescript
import { estimateTokens } from './utils.js'; // source file is utils.ts
```

The import resolver tried `utils.js` directly and appended more extensions (`utils.js.ts`, etc.), but never stripped `.js` to try `.ts`. Result: 0 call edges for any ESM project.

## Fix

In `resolveImportPath` (TS/JS relative path branch only):
1. If direct resolution fails and the path ends with a JS-family extension (`.js`/`.jsx`/`.mjs`/`.cjs`), strip it and re-resolve — matching TypeScript compiler behavior.
2. Added `.mts`/`.cts` to the EXTENSIONS list.

The fallback is scoped to TypeScript/JavaScript only — no impact on other languages.

## Testing

- 15 new unit tests covering all extension mappings, priority (real .js wins), non-TS languages, and edge cases.
- Manual verification: indexed a minimal ESM repo with `./utils.js` import → `context estimateTokens` correctly shows the caller.

## Verification

```
$ gitnexus context --repo esm-test estimateTokens
{
  "status": "found",
  "symbol": { "name": "estimateTokens", "filePath": "src/utils.ts" },
  "incoming": { "calls": [{ "name": "index.ts", "filePath": "src/index.ts" }] }
}
```

### CODE REVIEW

**Comment by magyargergo** [comment]
@chouzz Could you please look into claude's findings? 🙏

**Comment by chouzz** [comment]
Thanks @magyargergo! I've addressed all of Claude's findings:

  Fixed in this update:

  1. Finding 3 (makeCtx normalization) — Changed makeCtx to use .replace(/\\/g, '/') matching
  production buildImportResolutionContext behavior instead of .toLowerCase().
  2. Finding 1 (.mjs/.cjs precedence) — Added 3 tests documenting the resolution order when competing
  .ts/.mts siblings exist. The global EXTENSIONS order (.ts before .mts) is intentional — in practice,
  having both config.ts and config.mts in the same directory is extremely rare.
  3. Finding 2 (directory index boundary) — Added 2 tests documenting that ./dir.js → dir/index.ts is
  accepted behavior (bundler-mode compatibility, consistent with existing non-ESM resolution).
  4. Finding 4 (end-to-end CALLS edges) — Added an integration test
  (typescript-esm-js-extension.test.ts) that runs the full pipeline on a minimal ESM repo and asserts
  both the CALLS edge (processText → estimateTokens) and the IMPORTS edge are emitted correctly.
  5. Finding 5 (path alias + .js) — Added a code comment documenting this as a known limitation. The
  tsconfig alias branch does not yet apply stripJsExtension fallback. I'll file a separate issue for
  this as it expands scope beyond #1503.

  Follow-up issue to file:

  - Path alias imports with .js extensions (e.g. @/utils.js via tsconfig paths) do not trigger the ESM
  fallback. This requires applying the same strip-and-retry logic in the alias resolution branch. Will
  open a separate PR for this.

File: gitnexus/src/core/ingestion/import-resolvers/standard.ts [hunk 1]
```diff
@@ -132,6 +132,9 @@ export const resolveImportPath = (
 
     // TypeScript ESM: imports use .js/.jsx/.mjs/.cjs but source files are
     // .ts/.tsx/.mts/.cts. Strip the JS-family extension and re-resolve.
+    // NOTE: This fallback only applies to relative imports. Path alias imports
+    // (e.g. @/utils.js via tsconfig paths) do not yet strip .js extensions —
+    // that is a known limitation tracked for follow-up.
```
Comment (magyargergo) [inline gitnexus/src/core/ingestion/import-resolvers/standard.ts:137]: Please create a ticket for this

**Comment by chouzz** [comment]
Created #1528 to track the path alias + `.js` extension limitation. Thanks for the review!

**Comment by magyargergo** [comment]
> Created #1528 to track the path alias + `.js` extension limitation. Thanks for the review!

Are you happy to work on it?

**Comment by chouzz** [comment]
> > Created #1528 to track the path alias + `.js` extension limitation. Thanks for the review!
> 
> Are you happy to work on it?

Sure, you can assign to me, I will fix it.
